File: ryan-howard/WhoLikesIt.js
function likes(names) {                      //upvotes is the variable that counts the likes
  var upvotes = names.length;              //if statements determine output based on number of names. (Probably couldve just used names.length in hindsight)
                                              //each is just a string concatenation based on number of upvotes variable or number of names or names.length
  if (upvotes === 0) {
    return "no one likes this";
  } 
  if (upvotes === 1) {
    return names[0] + " likes this";
  }
  if (upvotes === 2){
    return names[0] + " and " + names[1] + " like this";
  }
  if (upvotes ===3){
    return names[0] + ", " + names[1] + " and " + names[2]+ " like this";
  }
  if (upvotes >= 4){
    return names[0] + ", " + names[1] + " and " + (upvotes -2) + " others like this";
  }
    
}

File: matthew-toffey/whoLikesit.py
def likes(names):
    output = ""
    if (len(names) == 0):
        output = "no one likes this"
    elif (len(names) == 1):
        output = str(names[0]) + " likes this"
    elif (len(names) == 2):
        output = str(names[0]) + " and " + str(names[1]) + " like this"
    elif (len(names) == 3):
        output = str(names[0]) + ", " + str(names[1]) + " and " + str(names[2]) + " like this"
    elif (len(names) >= 4):
        output = str(names[0]) + ", " + str(names[1]) + " and " + str(len(names) - 2) + " others like this"
    return output
    #begin
    #given a list of names based on liks
    #based on the names arr return name + "likes this" or "no one likes this"
    #if there are 4 or more names return the first 2 and "and # others like this"
    #end
